### DISCUSSION TITLE
DEMO does not populate any consent_events into GTM datalayer

### DISCUSSION THREAD

**Discussion by horde45** [orestbida/cookieconsent#646] [Q&A] [answered]
Hello, I downloaded the complete packages v 2.6 and 3.0, followed the instructions, put all necessary css and js files into the same directory on a web server, adjusted paths in index.html - and I stil do not see any consent events in GTM debug view . is that normal? I expected that upon confirming my cookie choices the script behind the banner will send consents events to datalayer where they would be grabbed by GTM console.

**Comment by orestbida** [comment]
There are no services pre configured with the plugin. You need to manually set up GTM.

**Reply by horde45** [reply]
many thanks for a quick reply.. Im not really into programing so please forgive me my stupid question:  do I get it correctly that on a datalayer level your script does not send anything to GTM that I could "grab" and use?

**Reply by orestbida** [reply, marked as answer]
Yes, this plugin is merely a skeleton. It does not implement any kind of service, but gives you the tools to easily set them up. You need to have basic knowledge of javascript to comfortably use the plugin.

[Here](https://github.com/orestbida/cookieconsent/discussions/634) is an example config. of GTM with google consent mode v2.

**Reply by horde45** [reply]
thank you for clarification.. and I spent hours trying to figure out why there are no consent events visible in GTM debug view..  perhaps it could be beneficial to include this information into readme file

**Reply by horde45** [reply]
I tried your hint with the on-page JS but it does not work as expected.

GA4 script part works and correctly sends consents into GTM datalayer

<script
    type="text/plain"
    data-category="analytics"
    data-service="Google Analytics 4">

    gtag('js', new Date());
    gtag('config', 'G-XXXXXXXXXX');

    gtag('consent', 'update', {
        personalization_storage: 'granted',
        functionality_storage: 'granted',
        analytics_storage: 'granted'
    });
    console.log("gtag: analytics_storage=granted");
</script>

---- BELOW DOES NOT send a %!@ to GTM datalayer - no consent granted

<script
    type="text/plain"
    data-category="ads"
    data-service="Google Ads">

    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}

    gtag('consent', 'update', {
        ad_storage: 'granted'
    });

    console.log("gtag: ad_storage=granted");
</script>
![240225 consent](https://github.com/orestbida/cookieconsent/assets/160885873/7c340c90-a661-479e-baf1-ceee1bfbd53b)

**Comment by horde45** [comment]
tried your hint with JS on-page script and it still does not work as intended. What am I doing wrong? 

`
GA4 script part works and correctly sends consents into GTM datalayer

<script
    type="text/plain"
    data-category="analytics"
    data-service="Google Analytics 4">

    gtag('js', new Date());
    gtag('config', 'G-XXXXXXXXXX');

    gtag('consent', 'update', {
        personalization_storage: 'granted',
        functionality_storage: 'granted',
        analytics_storage: 'granted'
    });
    console.log("gtag: analytics_storage=granted");
</script>

---- BELOW DOES NOT send a %!@ to GTM datalayer - no consent granted

<script
    type="text/plain"
    data-category="ads"
    data-service="Google Ads">

    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}

    gtag('consent', 'update', {
        ad_storage: 'granted'
    });

    console.log("gtag: ad_storage=granted");
</script>`
